### PR TITLE
[DEV-24149] Coalesce content cache misses and bound API fallback

### PR DESCRIPTION
**Pull request by camilb**
Concurrent Bea requests create separate ContentDelivery clients, so a cold cache currently sends the same work to the API once per request. In the local adapter reproduction, this change reduces a 100-request cold burst from 100 origin calls to 1; a warm burst makes 0 calls.

Cached content operations now share pending work across adapter instances in one JavaScript runtime, with source/token/header, storage, newsroom/theme/format, argument and version isolation. Scoped cache values prevent one authorization identity from consuming another's warm result. Successful responses return immediately; completed entries remain shared until their cache write settles or one second elapses. Late write failures are observed, including for custom caches whose writes outlive the retention deadline. Custom fetch implementations opt in with an explicit requestScope.

The cached content transport admits 32 active requests and 256 queued requests, with a 1s queue deadline and a 30s request deadline through body consumption. allStories produces eight pages at a time so large newsrooms cannot overflow that budget by themselves. Redis uses a 1s command deadline, a 128-command cap, and no offline command queue. The deadline deliberately avoids node-redis 4's unsafe cancellation of commands already sent to the socket. No additional API retries are introduced.

Validation:
- 37 new regression cases cover independent adapters, authorization/source isolation, cache writes (including never-settling custom writes), failure/retry, invalidation, admission limits, cancellation and large pagination. Existing package tests, builds, ESM/CJS import checks and declaration checks pass.
- Real loopback HTTP test confirms a stalled response body is aborted and a subsequent request succeeds.
- Isolated Redis test covers 100-request cold/warm bursts, 300 stalled commands, disconnect/reconnect, invalidation and stalled-cache fallback.
- Packed packages pass Bea's typecheck and production build with its SDK 26.5.0, plus an adapter smoke test. The build reports the Redis package's Edge-runtime warning; dependency installation reports the existing SDK peer-range mismatch.
- Independent review found the large-newsroom fan-out edge case; its repair passed the reviewer's 300-page reproduction and follow-up review (review-db 50/51). The quality-bot cache-write retention fix passed an additional independent review (review-db 52).

Release the changed core and Next.js packages together, then update Bea to the published version and deploy gradually. The fixed scoped-v1 namespace creates a one-time cold cache. This PR does not publish packages or update Bea's deployed dependency. Query canonicalization is tracked separately in DEV-24152; TTL and negative-cache policy remain follow-up work.

### CODE REVIEW

File: packages/core/src/content-delivery/RequestCoalescer.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        });
+        // Register before invoking, including synchronous errors and cache hits.
+        this.pending.set(key, result);
+        void (async () => {
+            try {
+                const { value, cacheWrite } = await invoke();
+                resolve(value);
+                // Followers can already use the value while the cache commits.
+                // Keep ownership to avoid another miss during a slow write.
+                await cacheWrite;
```
[the remark this answers is not in the document]
Comment (camilb) [reply packages/core/src/content-delivery/RequestCoalescer.ts]: Fixed in c0b4672482d64c83dee33296abdc2c3ec60d948f. Confirmed that an arbitrary Cache write could permanently retain both a resolved value and its pending slot.

Completed-result ownership now ends when the write settles or after one second, whichever comes first. Successful responses still return immediately. The timer is cleared on settlement, and a write that rejects after the deadline remains observed; late completion cannot remove a newer owner of the same key. This bounds coalescer ownership without assuming the custom Cache supports cancellation.

Added four regression cases for never-settling writes exhausting slots, late resolution/rejection while a newer owner is active, and real ContentDelivery clients rechecking cache/origin after the deadline. All four fail on the previous commit and pass with the fix. The package suites, dependent builds, ESM/CJS/declaration checks, focused test typecheck and Biome pass.
